examples/usages/context_menu.rs

```diff
@@ -0,0 +1,190 @@
+        let Ok(children) = children.get(trigger.target()) else {
+            return;
+        };
+
+        for child in children.iter() {
+            if let Ok(mut col) = query.get_mut(child) {
+                col.0 = color.into();
+            }
+        }
+    }
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2d);
+
+    commands.spawn(button()).observe(
+        // any click bubbling up here should lead to closing any open menu
+        |_trigger: Trigger<Pointer<Pressed>>, mut commands: Commands| {
+            commands.trigger(CloseContextMenus);
+        },
+    );
+}
+
+fn on_trigger_close_menus(
+    _trigger: Trigger<CloseContextMenus>,
+    mut commands: Commands,
+    menus: Query<Entity, With<ContextMenu>>,
+) {
+    for e in menus.iter() {
+        commands.entity(e).despawn();
+    }
+}
+
+fn on_trigger_menu(trigger: Trigger<OpenContextMenu>, mut commands: Commands) {
+    commands.trigger(CloseContextMenus);
+
+    let pos = trigger.pos;
+
+    debug!("open context menu at: {pos}");
+
+    commands
+        .spawn((
+            Name::new("context menu"),
+            ContextMenu,
+            Node {
+                position_type: PositionType::Absolute,
+                left: Val::Px(pos.x),
+                top: Val::Px(pos.y),
+                flex_direction: FlexDirection::Column,
+                ..default()
+            },
+            BorderColor(Color::BLACK),
+            BorderRadius::all(Val::Px(4.)),
+            BackgroundColor(Color::linear_rgb(0.1, 0.1, 0.1)),
+            children![
+                context_item("fuchsia", basic::FUCHSIA),
+                context_item("gray", basic::GRAY),
+                context_item("maroon", basic::MAROON),
+                context_item("purple", basic::PURPLE),
+                context_item("teal", basic::TEAL),
+            ],
+        ))
+        .observe(
+            |trigger: Trigger<Pointer<Pressed>>,
+             query: Query<&ContextMenuItem>,
+             mut clear_col: ResMut<ClearColor>,
+             mut commands: Commands| {
+                let target = trigger.event().target;
+
+                if let Ok(item) = query.get(target) {
+                    clear_col.0 = item.0.into();
+                    commands.trigger(CloseContextMenus);
+                }
+            },
+        );
+}
+
+fn context_item(text: &str, col: Srgba) -> impl Bundle + use<> {
+    (
+        Name::new(format!("item-{}", text)),
+        ContextMenuItem(col),
+        Button,
+        Node {
+            justify_content: JustifyContent::Center,
+            padding: UiRect::all(Val::Px(5.0)),
+            ..default()
+        },
+        children![(
+            Pickable::IGNORE,
+            Text::new(text),
+            TextFont {
+                font_size: 24.0,
+                ..default()
+            },
+            TextColor(Color::WHITE),
+        )],
+    )
+}
+
+fn button() -> impl Bundle + use<> {
+    (
+        Name::new("button"),
```

Comment:
This is not really a button anymore, just the background.

Reply:
correct, renamed